src/KafkaW/MetaDataQueryImpl.cpp

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: BSD-2-Clause
+//
+// This code has been produced by the European Spallation Source
+// and its partner institutes under the BSD 2 Clause License.
+//
+// See LICENSE.md at the top level for license information.
+//
+// Screaming Udder!                              https://esss.se
+
+#include "KafkaW/MetaDataQueryImpl.h"
+#include <algorithm>
+
+namespace KafkaW {
+const RdKafka::TopicMetadata *
+findKafkaTopic(const std::string &Topic,
+               const RdKafka::Metadata *KafkaMetadata) {
+  auto Topics = KafkaMetadata->topics();
+  auto Iterator =
+      std::find_if(Topics->cbegin(), Topics->cend(),
+                   [Topic](const RdKafka::TopicMetadata *TopicMetadata) {
+                     return TopicMetadata->topic() == Topic;
+                   });
+  if (Iterator == Topics->end()) {
+    throw MetadataException("Topic \"" + Topic + "\" not listed by broker.");
```

Comment:
Don't we use fmt to avoid this sort of thing?

Reply:
Yes, at least for log messages where the fmt functionality is  automatically included. In this case I could not be bothered to use the fmt-library as it would not have reduced the number of lines of code and I considered this text easy enough to read anyway.